server/api/src/rs_get_resource_info_for_operation.cpp

```diff
@@ -49,14 +49,16 @@ int rs_get_resource_info_for_operation(
                                   irods::WRITE_OPERATION,
                                   irods::UNLINK_OPERATION,
                                   irods::OPEN_OPERATION    };
-
-    decltype(allowed_ops.end()) matched_operation;
-    if (!op_type || allowed_ops.end() == (matched_operation = std::find(allowed_ops.begin(), allowed_ops.end(), op_type))) {
+    if (op_type == nullptr) {
+        return SYS_INVALID_INPUT_PARAM;
+    }
+    const auto *const found_operation = std::find(allowed_ops.begin(), allowed_ops.end(), op_type);
```

Comment:
`std::find` returns an iterator, not a pointer. So, I don't think we are going to want `const auto *const`, but rather just `const auto`.

Reply:
@d-w-moore - Y'know, now clang-tidy is saying that what you had was correct. Maybe my understanding of the interaction between auto specifiers and iterators is off...